src/resolver.rs

```diff
@@ -19,27 +20,44 @@ pub trait Resolver {
 /// a thread to provide non-blocking address resolving.
 #[derive(Clone)]
 pub struct CpuPoolResolver {
-    pool: CpuPool,
+    pool: futures::executor::ThreadPool,
 }
 
 impl CpuPoolResolver {
     /// Create a new CpuPoolResolver with the given number of threads.
-    pub fn new(num_threads: usize) -> Self {
+    pub fn new(num_threads: NonZeroUsize) -> Self {
```

Comment:
Why did you choose to use NonZeroUsize? I get that the pool cannot have 0 threads, but I'd rather keep the interface more ergonomic and just panic if 0 is used. Users are probably gonna do `NonZeroUsize::new(..).unwrap()` anyway.

Reply:
My reasoning is that all invariants should be moved into the type system as much as possible. Also I'm sure that in the real use this value will come verbatim from configuration/environment and you can directly deserialize/parse `NonZero` so it should not have ergonomics impact.

Reply:
I tried `Into/TryInto<NonZeroUsize>` but it doesn't seem like the trait has been implemented yet, bummer.